rlberry/envs/classic_control/SpringCartPole.py

```diff
@@ -0,0 +1,590 @@
+
+        pole1 = bar_shape(p01, p1, 0.01 * SCALE)
+        pole1.set_color((255 / 255, 215 / 255, 0 / 255))
+
+        pole2 = bar_shape(p02, p2, 0.01 * SCALE)
+        pole2.set_color((255 / 255, 215 / 255, 0 / 255))
+
+        spring = bar_shape(
+            p01,
+            p02,
+            0.03
+            * np.sqrt(self.normal_spring_length)
+            * SCALE
+            / np.sqrt(cartx2 - cartx1),
+        )
+        spring.set_color((50 / 255, 50 / 255, 50 / 255))
+
+        joint1 = circle_shape(p01, 0.03)
+        joint1.set_color((0 / 255, 255 / 255, 0 / 255))
+
+        joint2 = circle_shape(p02, 0.03)
+        joint2.set_color((0 / 255, 255 / 255, 0 / 255))
+
+        track_line = GeometricPrimitive("LINES")
+        track_line.add_vertex((-self.track_length / 2 * SCALE, -0.02 * SCALE))
+        track_line.add_vertex((self.track_length / 2 * SCALE, -0.02 * SCALE))
+
+        axis1 = GeometricPrimitive("LINES")
+        axis1.add_vertex((cartx1, 0))
+        axis1.add_vertex((cartx1, self.pole_length * SCALE))
+        axis1.set_color((250 / 255, 250 / 255, 250 / 255))
+
+        axis2 = GeometricPrimitive("LINES")
+        axis2.add_vertex((cartx2, 0))
+        axis2.add_vertex((cartx2, self.pole_length * SCALE))
+        axis2.set_color((250 / 255, 250 / 255, 250 / 255))
+
+        scene.add_shape(cart1)
+        scene.add_shape(cart2)
+        scene.add_shape(pole1)
+        scene.add_shape(pole2)
+        scene.add_shape(joint1)
+        scene.add_shape(joint2)
+        scene.add_shape(spring)
+        scene.add_shape(track_line)
+
+        return scene
+
+
+def wrap(x, m, M):
+    """Wraps ``x`` so m <= x <= M; but unlike ``bound()`` which
+    truncates, ``wrap()`` wraps x around the coordinate system defined
+    by m, M.
+    For example, m = -180, M = 180 (degrees), x = 360 --> returns 0.
+
+    Parameters
+    ----------
+        x: a scalar
+        m:
+            minimum possible value in range
+        M:
+            maximum possible value in range
+
+    Returns
+    -------
+        x:
+            a scalar, wrapped
+    """
+    diff = M - m
+    while x > M:
+        x = x - diff
+    while x < m:
+        x = x + diff
+    return x
+
+
+def bound(x, m, M=None):
+    """Either have m as scalar, so bound(x,m,M) which returns m <= x <= M *OR*
+    have m as length 2 vector, bound(x,m, <IGNORED>) returns m[0] <= x <= m[1].
+
+    Parameters
+    ----------
+    x:
+        scalar
+
+    Returns
+    -------
+    x:
+        scalar, bound between min (m) and Max (M)
+    """
+    if M is None:
+        M = m[1]
+        m = m[0]
+    # bound x between min (m) and Max (M)
+    return np.clip(x, m, M)
+
+
+def rk4(derivs, y0, t, *args, **kwargs):
+    """
+    Integrate 1D or ND system of ODEs using 4-th order Runge-Kutta.
+    This is a toy implementation which may be useful if you find
+    yourself stranded on a system w/o scipy.  Otherwise use
+    :func:`scipy.integrate`.
```

Comment:
scipy is a dependency of rlberry. 
Maybe you could use scipy here instead of doing it by hand ?